### PR TITLE
add unit tests

### PR DESCRIPTION
**Pull request by eiais**

### CODE REVIEW

**Comment by jqln-0** [comment]
yaml

yes,
*all*
my
lovers

File: vendor/github.com/davecgh/go-spew/LICENSE [hunk 1]
```diff
@@ -0,0 +1,15 @@
+ISC License
+
+Copyright (c) 2012-2016 Dave Collins <dave@davec.name>
```
Comment (jqln-0) [inline vendor/github.com/davecgh/go-spew/LICENSE]: delete this, fuck copyright

File: vendor/github.com/pmezard/go-difflib/LICENSE [hunk 2]
```diff
@@ -0,0 +1,27 @@
+Copyright (c) 2013, Patrick Mezard
```
Comment (jqln-0) [inline vendor/github.com/pmezard/go-difflib/LICENSE]: i wrote this now

File: vendor/github.com/stretchr/testify/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2012-2020 Mat Ryer, Tyler Bunnell and contributors.
```
Comment (jqln-0) [inline vendor/github.com/stretchr/testify/LICENSE]: i wrote this too. please delete this libelous text.

File: vendor/gopkg.in/yaml.v3/LICENSE [hunk 4]
```diff
@@ -0,0 +1,50 @@
+
+This project is covered by two different licenses: MIT and Apache.
```
Comment (jqln-0) [inline vendor/gopkg.in/yaml.v3/LICENSE]: 'apache'?

no thank u, no cultural appropriation here

File: vendor/gopkg.in/yaml.v3/NOTICE [hunk 5]
```diff
@@ -0,0 +1,13 @@
+Copyright 2011-2016 Canonical Ltd.
```
Comment (jqln-0) [inline vendor/gopkg.in/yaml.v3/NOTICE:1]: i don't like the orange and purple colour scheme these lads use

**Review by jqln-0 — CHANGES_REQUESTED** [review]
```   ,_____ ,
  ,._ ,_. 7\
 j `-'     /
 |o_, o    \
.`_y_`-,'   !
|/   `, `._ `-,
|_     \   _.'*\
  >--,-'`-'*_*'``---.
  |\_* _*'-'         '
 /    `               \
 \.         _ .       /
  '`._     /   )     /
   \  |`-,-|  /c-'7 /
    ) \ (_,| |   / (_
   ((_/   ((_;)  \_)))       ```

**Comment by eiais** [comment]
arf arf no licenses here

**Review by jqln-0 — APPROVED** [review]
good puppy

**Pull request merged**